Console/Command/CakePHPUtilShell.php

```diff
@@ -0,0 +1,330 @@
+	 *
+	 * @return void
+	 */
+	protected function _stop($code = 0) {
+		# stop the timer
+		$this->_out('TIME TO COMPLETE TASK: ' . date('H\h i\m s\s', microtime(true) - $this->_startTime - 3600));
+		$this->_printWarnings();
+		// Set arg names
+		$trace = $this->_getTrace();
+		$component = ($this->command) ? $this->command : 'Unknown';
+		$args = $this->args;
+		if (!isset($this->command) || !$this->hasMethod($this->command)) {
+			# if the command is not found, or it does not exist, shift the arguments by one, because it will remain on position 0 until it has ran.
+			array_shift($args);
+		}
+		$action = implode(' ', $args);
+		$this->Mail->send($this->_mailTo, $trace[1]['class'], $component, $action, $code, $trace[0]['line'], $this->_exceptions);
+
+		// Output result to console
+		if ($code > 0 && !empty($trace['line']) && !empty($this->_exceptions)) {
+			// Unexpected shutdown, with line specification and Exceptions
+			$this->out('The ' . $trace[1]['class'] . ' has unexpectedly terminated the \'' . $component . '/' . $action . '\' action with exit code ' . $code . '. The error occurred on line ' . $trace[0]['line'] . ' of the shell script. Additionally, the following Exceptions were thrown: ' . "\n\n" . implode("\r\n", $this->_exceptions));
+		} elseif ($code > 0 && !empty($trace['line'])) {
+			// Unexpected shutdown, with line specification
+			$this->out('The ' . $trace[1]['class'] . ' has unexpectedly terminated the \'' . $component . '/' . $action . '\' action with exit code ' . $code . '. The error occurred on line ' . $trace[0]['line'] . ' of the shell script.');
+		} elseif (!empty($this->_exceptions)) {
+			// Expected shutdown with Exceptions
+			$this->out('The ' . $trace[1]['class'] . ' has terminated the \'' . $component . '/' . $action . '\' action with exit code ' . $code . '. The error occurred on line ' . $trace[0]['line'] . '. Additionally, the following Exceptions were thrown: ' . "\n\n" . implode("\r\n", $this->_exceptions));
+		} else {
+			$this->out(sprintf('The %s has ran successfully. No exceptions were thrown.', $trace[1]['class']));
+		}
+		$this->hr();
+
+		// Exit the Shell
+		parent::_stop($code);
+	}
+
+/**
+ * Print warnings that were catched by the AppShellErrorHandler.
+ */
+	protected function _printWarnings() {
+		if (AppShellErrorHandler::hasWarnings()) {
+			$warnings = AppShellErrorHandler::flushWarnings();
+			$this->out("The following warnings occurred during the cronjob:\n");
+			foreach ($warnings as $warning) {
+				$this->out(trim($warning[1]));
+				foreach ($warning[2] as $backtrace) {
+					// The warning_handler doesn't have a a file and line number in the backtrace.
+					if (!isset($backtrace['line']) && !isset($backtrace['file'])) {
+						continue;
+					}
+					$this->out($backtrace['class'] . $backtrace['type'] . $backtrace['function'] . ' in ' . $backtrace['file'] . ' on line ' . $backtrace['line']);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this 148 characters long line (which is greater than 120 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://dev.expandonline.nl:9000/sonarqube/coding_rules#rule_key=php%3AS103)